src/dodal/beamlines/i22.py

```diff
@@ -23,27 +29,81 @@
 from dodal.log import set_beamline as set_log_beamline
 from dodal.utils import BeamlinePrefix, get_beamline_name, skip_device
 
-BL = get_beamline_name("i22")
+LAB_NAME = BEAMLINE_LAB_MAPPING.get("i22", "i22")
+
+BL = LAB_NAME if LAB_FLAG else get_beamline_name("i22")
+print("BL NAME: ", BL)
 set_log_beamline(BL)
 set_utils_beamline(BL)
 
+IS_LAB = LAB_FLAG and BL == LAB_NAME
+print("is this lab? : ", IS_LAB)
+
 # Currently we must hard-code the visit, determining the visit at runtime requires
 # infrastructure that is still WIP.
 # Communication with GDA is also WIP so for now we determine an arbitrary scan number
 # locally and write the commissioning directory. The scan number is not guaranteed to
 # be unique and the data is at risk - this configuration is for testing only.
-set_directory_provider(
+directory_provider = (
```

Comment:
```suggestion
set_directory_provider(
```

We need to call this method for the StandardDetectors to write to disk

Reply:
it is already called later